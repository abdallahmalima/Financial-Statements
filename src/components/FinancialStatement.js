import PropTypes from 'prop-types';
import { useNavigate } from 'react-router-dom';

const FinancialStatement = ({
  symbol,
}) => {
  const navigate = useNavigate();

  const showIncomeStatements = () => {
    navigate(`/${symbol}`);
  };
  return (
<div className="w-1/2 px-2 mb-4">
      <div className="bg-red-500 rounded-lg shadow-lg overflow-hidden">
        <div className="flex flex-col p-5 items-center">
          <h3 className="text-2xl font-bold text-gray-900 mb-2">{symbol}</h3>
          <button onClick={showIncomeStatements} className="px-2 py-2 font-semibold rounded shadow-lg bg-red-700 text-gray-300 text-xl">
  View More
</button>
        </div>
      </div>
    </div>
  );
};

FinancialStatement.propTypes = {
  symbol: PropTypes.string.isRequired,
};

export default FinancialStatement;