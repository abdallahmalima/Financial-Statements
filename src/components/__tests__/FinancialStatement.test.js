import { render } from '@testing-library/react';
import { HashRouter as Router } from 'react-router-dom';
import FinancialStatement from '../FinancialStatement';

describe('Test Financial Statement Component', () => {
  it('check if component component match the snapshot', () => {
    const symbol = 'ABC';

    const financialStatementRender = render(
        <Router>
        <FinancialStatement symbol={symbol} />
        </Router>,
    );

    expect(financialStatementRender).toMatchSnapshot();
  });

  it('renders symbol and button', () => {
    const symbol = 'ABC';
    const { getByText } = render(
      <Router>
       <FinancialStatement symbol={symbol} />
    </Router>,
    );
    const textSymbol = getByText(/ABC/i);
    const buttonElement = getByText(/view more/i);

    expect(buttonElement).toBeInTheDocument();
    expect(textSymbol).toBeInTheDocument();
  });
});
