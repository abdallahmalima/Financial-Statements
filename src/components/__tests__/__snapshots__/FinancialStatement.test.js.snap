// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Financial Statement Component check if component component match the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="w-1/2 px-2 mb-4"
      >
        <div
          class="bg-red-500 rounded-lg shadow-lg overflow-hidden"
        >
          <div
            class="flex flex-col p-5 items-center"
          >
            <h3
              class="text-2xl font-bold text-gray-900 mb-2"
            >
              ABC
            </h3>
            <button
              class="px-2 py-2 font-semibold rounded shadow-lg bg-red-700 text-gray-300 text-xl"
            >
              View More
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="w-1/2 px-2 mb-4"
    >
      <div
        class="bg-red-500 rounded-lg shadow-lg overflow-hidden"
      >
        <div
          class="flex flex-col p-5 items-center"
        >
          <h3
            class="text-2xl font-bold text-gray-900 mb-2"
          >
            ABC
          </h3>
          <button
            class="px-2 py-2 font-semibold rounded shadow-lg bg-red-700 text-gray-300 text-xl"
          >
            View More
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
